src/sentry/utils/cloudfunctions.py

```diff
@@ -0,0 +1,153 @@
+from io import BytesIO
+from zipfile import ZipFile
+
+import requests
+from google.cloud.functions_v1.services.cloud_functions_service import CloudFunctionsServiceClient
+from google.cloud.functions_v1.types import (
+    CloudFunction,
+    EventTrigger,
+    GenerateUploadUrlRequest,
+    UpdateFunctionRequest,
+)
+from google.cloud.functions_v1.types.functions import DeleteFunctionRequest
+from google.cloud.pubsub_v1 import PublisherClient
+from google.protobuf.field_mask_pb2 import FieldMask
+
+from sentry.utils import json
+
+WRAPPER_JS = """
```

Comment:
Basically, this is the code that we put into Google Cloud Function alongside user defined code. It monitors the user created function with Sentry so that when something is wrong we can notify the user.